core/block.ts

```diff
@@ -1,341 +1,269 @@
+  isInMutator: boolean;
+  RTL: boolean;
 
-    /**
-     * A string representing the comment attached to this block.
-     * @type {string|Comment}
-     * @deprecated August 2019. Use getCommentText instead.
-     */
-    this.comment = null;
+  /** True if this block is an insertion marker. */
+  protected isInsertionMarker_ = false;
 
-    /**
-     * A model of the comment attached to this block.
-     * @type {!Block.CommentModel}
-     * @package
-     */
-    this.commentModel = {text: null, pinned: false, size: new Size(160, 80)};
+  /** Name of the type of hat. */
+  hat?: string = undefined;
 
-    /**
-     * The block's position in workspace units.  (0, 0) is at the workspace's
-     * origin; scale does not change this value.
-     * @type {!Coordinate}
-     * @private
-     */
-    this.xy_ = new Coordinate(0, 0);
+  rendered: boolean | null = null;
 
-    /** @type {!Workspace} */
-    this.workspace = workspace;
-    /** @type {boolean} */
-    this.isInFlyout = workspace.isFlyout;
-    /** @type {boolean} */
-    this.isInMutator = workspace.isMutator;
+  /**
+   * String for block help, or function that returns a URL. Null for no help.
+   */
+  // AnyDuringMigration because:  Type 'null' is not assignable to type 'string
+  // | Function'.
+  helpUrl: string | Function = null as AnyDuringMigration;
 
-    /** @type {boolean} */
-    this.RTL = workspace.RTL;
+  /** A bound callback function to use when the parent workspace changes. */
+  private onchangeWrapper_: ((p1: Abstract) => AnyDuringMigration) | null = null;
 
-    /**
-     * True if this block is an insertion marker.
-     * @type {boolean}
-     * @protected
-     */
-    this.isInsertionMarker_ = false;
+  /** A count of statement inputs on the block. */
+  statementInputCount = 0;
+  // TODO(b/109816955): remove '!', see go/strict-prop-init-fix.
+  type!: string;
+  // Record initial inline state.
+  inputsInlineDefault?: boolean;
+  workspace: Workspace;
 
-    /**
-     * Name of the type of hat.
-     * @type {string|undefined}
-     */
-    this.hat = undefined;
+  /**
+   * @param workspace The block's workspace.
+   * @param prototypeName Name of the language object containing type-specific
+   *     functions for this block.
+   * @param opt_id Optional ID.  Use this ID if provided, otherwise create a new
+   *     ID.
+   * @throws When the prototypeName is not valid or not allowed.
+   */
+  constructor(workspace: Workspace, prototypeName: string, opt_id?: string) {
+    // AnyDuringMigration because:  Property 'get' does not exist on type
+    // '(name: string) => void'.
+    const { Generator } =
+      (goog.module as AnyDuringMigration).get('Blockly.Generator');
+    if (Generator &&
+      typeof Generator.prototype[prototypeName] !== 'undefined') {
+      // Occluding Generator class members is not allowed.
+      throw Error(
+        'Block prototypeName "' + prototypeName +
+        '" conflicts with Blockly.Generator members.');
+    }
 
```

Comment:
I don't think this is going to work anymore (and maybe currently doesn't work?). My understanding is that the TS code won't have access to the `Blockly` namespace tree. So `Generator.prototype` won't ever have other stuff on it for this to conflict.

Needs @cpcallen 's understanding.